source/common/protobuf/utility.cc

```diff
@@ -505,6 +506,79 @@ std::string MessageUtil::CodeEnumToString(ProtobufUtil::error::Code code) {
   }
 }
 
+namespace {
+
+// Recursive helper method for MessageUtil::redact() below.
+void redactInPlace(Protobuf::Message* message, bool ancestor_is_redacted) {
+  // If the message is an `Any`, we can't use any type-based introspection. Instead, we have to
+  // unpack it to its original type to redact it...
+  auto* any = dynamic_cast<ProtobufWkt::Any*>(message);
+  if (any != nullptr) {
+    // 1. Extract the type name from the URL. This is guaranteed to be the portion of the URL
+    // following the last '/' character.
+    const std::string& type_url = any->type_url();
+    const std::size_t last_slash = type_url.find_last_of('/');
+    const std::string type_name = type_url.substr(last_slash + 1);
+    const auto* descriptor =
```

Comment:
What if descriptor is null? This might be true if we log during config update before we verify that we know about all the embedded opaque types.

Reply:
Hmm... I think I should just log a warning, like "could not load descriptor for type <type name>" in that case. WDYT?

Reply:
In theory this could lead to information leakage, since we might emit the opaque `Any` representation (not sure what this looks like under logging, but it could be a binary output as well). In practice, as long as we only redact around log messages, we're confident that there are no nested redactions and/or that proto descriptors are always known then we're good. 

If we only redact on Envoy's built-in protos this is likely safe. I feel we should have better  documentation about this gotcha somewhere.